### PR TITLE
ILM: Allow `searchable_snapshot` to follow `searchable_snapshot`

### PR DESCRIPTION
**Pull request by andreidan**
We added this validation for the hot phase as that was the only place a
searchable_snapshot action could precede another `searchable_snapshot` action.
This is not the case anymore and we now support multiple actions in the
same policy.

This changes the validation to allow the `searchable_snapshot` *in the hot
phase* to be followed by other `searchable_snapshot` actions.

### CODE REVIEW

File: x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/actions/SearchableSnapshotActionIT.java [hunk 1]
```diff
@@ -449,23 +450,32 @@ public void testConvertingSearchableSnapshotFromFullToPartial() throws Exception
...
     public void testConvertingPartialSearchableSnapshotIntoFull() throws Exception {
-        String index = "myindex-" + randomAlphaOfLength(4).toLowerCase(Locale.ROOT);
+        String index = "myindex-" + randomAlphaOfLength(4).toLowerCase(Locale.ROOT) +"-000001";
         createSnapshotRepo(client(), snapshotRepo, randomBoolean());
-        createPolicy(client(), policy, null, null,
-            new Phase("cold", TimeValue.ZERO,
-                singletonMap(SearchableSnapshotAction.NAME, new SearchableSnapshotAction(snapshotRepo, randomBoolean(),
-                    MountSearchableSnapshotRequest.Storage.SHARED_CACHE))),
+        createPolicy(client(), policy,
+            new Phase("hot", TimeValue.ZERO,
```
Comment (andreidan) [inline x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/actions/SearchableSnapshotActionIT.java:456]: Piggybacked the hot -> frozen searchable_snapshot actions on this test (as it wasn't tested anywhere)

**Comment by andreidan** [comment]
@elasticmachine update branch

**Comment by andreidan** [comment]
This is still a `WIP`. The validation currently looks at the `hot` phase specifically and I think it should be more general.

**Review by dakrone — DISMISSED** [review]
LGTM

**Comment by andreidan** [comment]
@elasticmachine update branch

**Review by dakrone — APPROVED** [review]
LGTM, thanks for fixing this

**Comment by andreidan** [comment]
@elasticmachine run elasticsearch-ci/2

**Pull request merged**